sherpa/astro/models/tests/test_astro_model.py

```diff
@@ -79,18 +79,12 @@ def test_create_and_evaluate(name, cls):
         assert out.shape == x.shape
 
 
-@pytest.mark.parametrize("cls", [models.Atten,
-                                 models.BBody,
-                                 models.BBodyFreq,
-                                 models.BPL1D,
+@pytest.mark.parametrize("cls", [
                                  models.Beta1D,
-                                 models.Edge,
                                  models.LineBroad,
                                  models.Lorentz1D,
-                                 models.NormBeta1D,
                                  models.PseudoVoigt1D,
-                                 models.Schechter,
-                                 models.Voigt1D])
```

Comment:
So, why do these models that you remove fail? Oh, I guess it's just that because they have no cache that the `len(mdl._cache)` checks are no-longer valid.

There was a reason I wrote this test. If you added a test like

```python
@pytest.mark.parametrize("cls", [models.Atten, models.Schechter])  # just pick a few
def test_send_keyword_1d_no_cache(cls):
    """These models do not use a cache. Can we send an unsupported keyword?"""

    mdl = cls()
    x = [10, 21, 13, 15]
    y1 = mdl(x)
    y2 = mdl(x, not_an_argument=True)
    assert len(mdl._cache) == 0
    assert y2 == pytest.approx(y1)
```

Oh, but we essentially have this in `test_send_keyword_1d_no_cache`, so there's no need. But I've already written this comment so I'm going to comment it....